crates/runtime-table-partition/src/provider.rs

```diff
@@ -118,6 +136,136 @@ impl TableProvider for PartitionTableProvider {
         _filters: &[Expr],
         _limit: Option<usize>,
     ) -> Result<Arc<dyn ExecutionPlan>, DataFusionError> {
-        todo!()
+        Err(DataFusionError::Execution(
+            "PartitionedTableProvider::scan not implemented yet".to_string(),
+        ))
+    }
+
+    async fn insert_into(
+        &self,
+        state: &dyn Session,
+        input: Arc<dyn ExecutionPlan>,
+        insert_op: InsertOp,
+    ) -> Result<Arc<dyn ExecutionPlan>, DataFusionError> {
```

Comment:
The insert_into method itself shouldn't do the computation - it should return an ExecutionPlan which represents the computation graph. If you look at the current implementations, you'll see that just return a `DataSinkExec` which actually handles the insertion.